pkg/util/lock/key_lock.go

```diff
@@ -118,15 +146,22 @@ func (k *KeyLock[K]) RUnlock(lockedKey K) {
 		log.Warn("Unlocking non-existing key", zap.Any("key", lockedKey))
 		return
 	}
-	keyLock.unref()
-	if keyLock.refCounter == 0 {
-		delete(k.refLocks, lockedKey)
+	success := keyLock.unref()
+	if !success {
+		log.Warn("Unlocking non-locked key", zap.Any("key", lockedKey))
```

Comment:
I guess we should panic here.

Reply:
Panic may not be a good idea, because the native lock/unlock function has linter support. For our custom lock function, we may unlock the lock multiple times. In theory, unlocking the lock multiple times will not have any impact. So I think it is acceptable to just warn the lock function.

Reply:
> we may unlock the lock multiple times. In theory

May I ask the scenarios?

Reply:
Wrong code, ahhhh